packages/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx

```diff
@@ -1059,4 +1052,31 @@ DataGridPremiumRaw.propTypes = {
    * @default false
    */
   treeData: PropTypes.bool,
+  unstable_dataSource: PropTypes.shape({
+    getRows: PropTypes.func.isRequired,
+    updateRow: PropTypes.func,
+  }),
+  unstable_onServerSideError: PropTypes.func,
+  unstable_serverSideCache: PropTypes.shape({
```

Comment:
Wdyt about naming these `dataSourceCache` and `onDataSourceError`? Keeps the names consistent and makes it super obvious that these props are all related.

Reply:
And by extension, we could also rename the `serverSideData` folder & state field to just `dataSource`? The other keys in the state just use the name of the feature without adding "data", it's a bit less verbose.

Reply:
That connects to another discussion I wanted to do.
That is, about the naming of the concept. Right now, we use `dataSource`, for reference, toolpad [names](https://mui.com/toolpad/studio/concepts/data-providers/) a similar thing on their side as `dataProvider` and there was another idea calling it `dataManager`.

This was the reason I was inclined to name the props more generic, since the name was not finalized. I am more aligned with both `dataSource` and `dataProvider`. It would be great to know which one feels better/more natural to you (CC @cherniavskii, @romgrk)

I'll refactor the naming afterward.

Reply:
I have a small preference for "data source".

Not sure what toolpad does, but I wouldn't use the same name unless it's API-compatible with their data provider object.

Reply:
Alright, I'll refactor the naming then 👍